src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2889,5 +2937,55 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Next
             End Sub
         End Class
+
+        Private Class PredicateSymbolSearcher
+            Inherits AbstractSymbolSearcher
+
+            Private ReadOnly _predicate As Func(Of String, Boolean)
+
+            Public Sub New(
+                compilation As VisualBasicCompilation, filter As SymbolFilter, predicate As Func(Of String, Boolean), cancellationToken As CancellationToken)
+                MyBase.New(compilation, filter, cancellationToken)
+
+                _predicate = predicate
+            End Sub
+
+            Protected Overrides Function ShouldCheckTypeForMembers(current As MergedTypeDeclaration) As Boolean
+                Return True
+            End Function
+
+            Protected Overrides Function Matches(name As String) As Boolean
+                Return _predicate(name)
+            End Function
+        End Class
+
+        Private Class NameSymbolSearcher
+            Inherits AbstractSymbolSearcher
+
+            Private ReadOnly _name As String
+
+            Public Sub New(
+                compilation As VisualBasicCompilation, filter As SymbolFilter, name As String, cancellationToken As CancellationToken)
+                MyBase.New(compilation, filter, cancellationToken)
+
+                _name = name
+            End Sub
+
+            Protected Overrides Function ShouldCheckTypeForMembers(current As MergedTypeDeclaration) As Boolean
+                For Each typeDecl In current.Declarations
+                    For Each name In typeDecl.MemberNames
+                        If IdentifierComparison.Equals(name, _name) Then
+                            Return True
+                        End If
+                    Next
+                Next
```

Comment:
note: it's a pity we can't just do typeDecl.MemberNames.Contains(_name)  However, MemberNames is just an array, and not a set with an appropriate comparer.

Reply:
We could potentially replace that array with a SmallDictionary.Keys, but that could potentially have larger remifications.  Personally, i don't htink it would be an issue, as this would be just one small change per TypeDeclaration, but it seemed out of scope of this PR.  @VSadov for his thoughts here.

Reply:
Note: i was the original person who added MemberNames to the decl table.  It was always intended, from the beginning, to be a fast way to check for existence of a member and/or to get that member quickly.  I believe i started things out with a set, but it's so lost in the annals of time for me that i'm not sure.  However, it does feel like this would be an easy tweak to make, with minimal issues, for the beneficial goal of speeding up these queries.  Being able to just do simple O(1) lookup per type, instead of N checks would be *very* nice.